underfs/s3a/src/main/java/alluxio/underfs/s3a/S3ALowLevelOutputStream.java

```diff
@@ -11,479 +11,184 @@
-        mKey, uploadId, mTags);
-    do {
-      try {
-        s3Client.completeMultipartUpload(completeRequest);
-        LOG.debug("Completed multipart upload for key {} and id '{}' with {} partitions.",
-            mKey, uploadId, mTags.size());
-        return;
-      } catch (AmazonClientException e) {
-        lastException = e;
+  @Override
+  protected void putObject(String key, File file, String md5) throws IOException {
+    try {
+      ObjectMetadata meta = new ObjectMetadata();
+      if (mSseEnabled) {
+        meta.setSSEAlgorithm(ObjectMetadata.AES_256_SERVER_SIDE_ENCRYPTION);
       }
-    } while (mRetryPolicy.attempt());
-    // This point is only reached if the operation failed more
-    // than the allowed retry count
-    throw new IOException("Unable to complete multipart upload with id '"
-        + uploadId + "' to " + mKey, lastException);
-  }
-
-  protected void abortMultiPartUpload() {
-    abortMultiPartUpload(getClient(), mUploadId);
-  }
-
-  /**
-   * Aborts multipart upload.
-   */
-  protected void abortMultiPartUpload(AmazonS3 s3Client, String uploadId) {
-    AmazonClientException lastException;
-    do {
-      try {
-        s3Client.abortMultipartUpload(new AbortMultipartUploadRequest(mBucketName,
-            mKey, uploadId));
-        LOG.warn("Aborted multipart upload for key {} and id '{}' to bucket {}",
-            mKey, uploadId, mBucketName);
-        return;
-      } catch (AmazonClientException e) {
-        lastException = e;
+      if (md5 != null) {
+        meta.setContentMD5(md5);
       }
-    } while (mRetryPolicy.attempt());
-    // This point is only reached if the operation failed more
-    // than the allowed retry count
-    LOG.warn("Unable to abort multipart upload for key '{}' and id '{}' to bucket {}. "
-        + "You may need to enable the periodical cleanup by setting property {}"
-        + "to be true.", mKey, uploadId, mBucketName,
-        PropertyKey.UNDERFS_CLEANUP_ENABLED.getName(),
-        lastException);
-  }
-
-  /**
-   * Validates the arguments of write operation.
-   *
-   * @param b the data
-   * @param off the start offset in the data
-   * @param len the number of bytes to write
-   */
-  private void validateWriteArgs(byte[] b, int off, int len) {
-    Preconditions.checkNotNull(b);
-    if (off < 0 || off > b.length || len < 0
-        || (off + len) > b.length || (off + len) < 0) {
-      throw new IndexOutOfBoundsException("write(b[" + b.length + "], " + off + ", " + len + ")");
+      meta.setContentLength(file.length());
+      meta.setContentType(Mimetypes.MIMETYPE_OCTET_STREAM);
+      PutObjectRequest putReq = new PutObjectRequest(mBucketName, key, file).withMetadata(meta);
+      mManager.upload(putReq).waitForUploadResult();
+    } catch (Exception e) {
+      throw new IOException(e);
     }
   }
-
-  /**
-   * @return the client
-   */
-  protected AmazonS3 getClient() {
```

Comment:
please keep this getClient function and use this function in the whole implementation since some Enterprise features would be based on this.

Reply:
fix.